packages/lesswrong/server/migrations/2020-08-09-metaToCommunity.ts

```diff
@@ -0,0 +1,160 @@
+          // that setting with the personalBlogpost setting that now might
+          // actually be about personal blogposts
+          if (user.frontpageFilterSettings?.tags?.some((setting: FilterTag) => setting.tagId === communityTagId))
+          {
+            return []
+          }
+          const communityWeight = user.frontpageFilterSettings?.personalBlog || defaultFilterSettings.personalBlog
+
+          return [
+            {
+              updateOne: {
+                filter: { _id: user._id },
+                update: {$push: {'frontpageFilterSettings.tags': {
+                  tagId: communityTagId,
+                  tagName: "Community",
+                  filterMode: communityWeight,
+                }}}
+              }
+            }, {
+              updateOne: {
+                filter: { _id: user._id },
+                update: {$set: {
+                  'frontpageFilterSettings.personalBlog': defaultFilterSettings.personalBlog
+                }}
+              }
+            }
+          ] as MongoModifier<DbUser>
+        })
+        
+        if (changes.length) await Users.rawCollection().bulkWrite(changes, { ordered: false })
+      }
+    })
+  }
+})
+
+// Someone's gotta put their name on all those tagRels
+const DEFAULT_ADMIN_USER_SLUG = 'jpaddison'
+
+// Tag all posts with the meta flag as community posts
+registerMigration({
+  name: 'metaToCommunityPosts',
+  dateWritten: '2020-08-12',
+  idempotent: true,
+  action: async () => {
+    const communityTagId = Tags.find({slug: 'community'}).fetch()[0]._id
+    const defaultAdminUserId = Users.find({ slug: DEFAULT_ADMIN_USER_SLUG }).fetch()[0]._id
+
+    await forEachDocumentBatchInCollection({
+      collection: Posts,
+      batchSize: 100,
+      filter: {meta: true},
+      callback: async (posts: Array<DbPost>) => {
+        // eslint-disable-next-line no-console
+        console.log("Migrating post batch");
+        
+        for (let post of posts) {
+          if (post.tagRelevance?.[communityTagId]) {
+            continue
+          }
+          // Oh man, I refactored this migration to use this method, and it
+          // changed my life. 10/10 would use again in future migrations.
+          // bulkwrite is faster, but callbacks are often important
+          await newMutation({
+            collection: TagRels,
+            document: {
+              tagId: communityTagId,
+              postId: post._id,
+              userId: post.reviewedByUserId || defaultAdminUserId,
+            },
+            // Validation requires us to have a context object, which has
+            // things like currentUser that aren't applicable.
+            validate: false,
+          })
+        }
+      }
+    })
+  }
+})
+
+// Once we've deployed, run this migration to mark all community posts as
+// frontpage, and remove the meta flag
+registerMigration({
+  name: 'moveMetaToFrontpage',
```

Comment:
Interested why you're doing this as two passes, given you're operating on exactly the same set of posts as in the migration above?